Sources/SWBCore/SpecImplementations/Tools/CCompiler.swift

```diff
@@ -1136,7 +1137,8 @@ public class ClangCompilerSpec : CompilerSpec, SpecIdentifierType, GCCCompatible
         //
         // FIXME: Eventually we should just apply this optimization to all generic specs, and then find a way to piggy back on that.
         let dataCache = cbc.producer.getSpecDataCache(self, cacheType: DataCache.self)
-        let constantFlags = dataCache.getStandardFlags(self, producer: cbc.producer, scope: cbc.scope, optionContext: clangInfo, delegate: delegate, inputFileType: resolvedInputFileType)
+        let cbcWithOutput = cbc.outputs.isEmpty ? cbc.appendingOutputs([outputNode.path]) : cbc
+        let constantFlags = dataCache.getStandardFlags(self, cbc: cbcWithOutput, optionContext: clangInfo, delegate: delegate, inputFileType: resolvedInputFileType)
```

Comment:
These flags are cached and reused across translation units, so I think this is going to incorrectly inject per-TU flags